packages/common/src/hooks/useImageSize2.ts

```diff
@@ -28,11 +19,13 @@ const preloadImage = (url: string) => {
 export const useImageSize2 = ({
   artwork,
   targetSize,
-  defaultImage
+  defaultImage,
+  preloadImageFn
 }: {
-  artwork?: Track['artwork']
+  artwork?: Track['artwork'] | Collection['artwork']
   targetSize: SquareSizes
   defaultImage: string
+  preloadImageFn: (url: string) => Promise<void>
```

Comment:
This _could_ just be provided via AppContext if you want. That's how we do most of the platform-specific injections.